Zend/zend_alloc.c

```diff
@@ -1317,7 +1360,7 @@ static zend_always_inline void *zend_mm_alloc_small(zend_mm_heap *heap, int bin_
 
 	if (EXPECTED(heap->free_slot[bin_num] != NULL)) {
 		zend_mm_free_slot *p = heap->free_slot[bin_num];
-		heap->free_slot[bin_num] = p->next_free_slot;
+		heap->free_slot[bin_num] = zend_mm_check_next_free_slot(heap, bin_num, p);
```

Comment:
`next_free_slot` is checked when moving from `p->next_free_slot` to `heap->free_slot[bin_num]` (and also in zend_mm_gc()).